python/pyarrow/array.pxi

```diff
@@ -339,8 +339,12 @@ def array(object obj, type=None, mask=None, size=None, from_pandas=None,
             result = _ndarray_to_array(values, mask, type, c_from_pandas, safe,
                                        pool)
     else:
+        if type and type.id == _Type_RUN_END_ENCODED:
+            from pyarrow.compute import run_end_encode
+            result = run_end_encode(obj)
```

Comment:
We should ensure that we actually honor the values type of the REE type. Eg if using `pa.run_end_encoded(run_end_type=pa.int32(), value_type=pa.float64())`, the result should use that exact type, and not whatever is inferred by `pc.run_end_encode`

Reply:
Makes sense!
Would it be a good approach to keep the compute function to actually get `.run_ends` and `.values` and then call `RunEndEncodedArray.from_arrays(run_ends, values, ree_type)` with the correct ree type? (it shouldn't make a copy at this point)